src/TextArea.js

```diff
@@ -0,0 +1,246 @@
+import propTypes from '@dhis2/prop-types'
+import React, { PureComponent } from 'react'
+import cx from 'classnames'
+
+import { statusPropType } from './common-prop-types.js'
+import { StatusIconNoDefault } from './icons/Status.js'
+import { detectScrollbarSize } from './Tabs/detectScrollbarSize.js'
+
+import { styles } from './TextArea/styles.js'
+
+export class TextArea extends PureComponent {
+    constructor(props) {
+        super(props)
+
+        this.textareaRef = React.createRef()
+
+        this.state = {
```

Comment:
I think all the properties could be declared as properties instead of in the constructor (they don't depend on props).
With the other change in the comment below, we could get rid of the constructor.